### ISSUE TITLE
Cannot read property 'map' of undefined

### ISSUE THREAD

**Issue by di97mni** [flant/grafana-statusmap#143]
Grafana 7.2.2
Statusmap : 0.3.4

Getting this error when upgrading to Grafana 7.2.2 and Statusmap 0.3.4
Any ideas what to look for?

![image](https://user-images.githubusercontent.com/311406/96754311-6942a500-13d1-11eb-9400-1aa50310a828.png)

![image](https://user-images.githubusercontent.com/311406/96754414-8e371800-13d1-11eb-8d01-c44cec9321a3.png)

![image](https://user-images.githubusercontent.com/311406/96754445-98591680-13d1-11eb-85c7-7981e3ab91e9.png)

![image](https://user-images.githubusercontent.com/311406/96754475-a0b15180-13d1-11eb-84f2-b6e2a4a56c68.png)

**Comment by diafour** [comment]
Hello. My first thought it is a 'no data' case. It is a known and long-lasting bug.

**Comment by di97mni** [comment]
The query, to log analytics in Azure, returns data.
This is what it looked like before the upgrade.

![image](https://user-images.githubusercontent.com/311406/96831677-c4aa7c80-143d-11eb-922b-39ecc7b5594b.png)

**Comment by di97mni** [comment]
Example of response

![image](https://user-images.githubusercontent.com/311406/96832325-db9d9e80-143e-11eb-9cdb-3258cd4cf3a8.png)

**Comment by diafour** [comment]
Yep, it is definitely not a "No data" issue. Unfortunately, I have no access to Log Analytics Workspace to reproduce the issue. I've tried Azure Monitor queries and it works as expected.

Can you please help gather some debug information to fix this issue? You need to navigate to a 'module.ts' in 'Sources' pane of DevTools and set a breakpoint on line 365. Then press 'reload' button in a dashboard. Next scroll to Scope and expand 'dataList' variable. 

Example 1 with Prometheus datasource with 2 queries:
<img width="1026" alt="Снимок экрана 2020-10-23 в 21 00 37" src="https://user-images.githubusercontent.com/1055474/97038820-105f4200-1574-11eb-88c7-7ab91f0d3217.png">

Example 2 with Azure Monitor also with 2 queries.
<img width="948" alt="Снимок экрана 2020-10-23 в 21 03 27" src="https://user-images.githubusercontent.com/1055474/97038842-181ee680-1574-11eb-86ec-f2c23331d278.png">

**Comment by di97mni** [comment]
Haven't forgot this but I haven't been able to test it again. Needed to roll back to 7.0.0. 
Will test it again as soon as I can

**Comment by shurup** [comment]
Hi @di97mni, are there any news on that?

**Comment by tambetliiv** [comment]
Had the same problem

- Grafana v7.3.7 (1e261642f4)
- Datasource https://github.com/alexanderzobnin/grafana-zabbix v4.1.1
- Statusmap v0.3.4

Console errors:
![image](https://user-images.githubusercontent.com/35329231/105864836-03327300-5ffb-11eb-8dea-568012fcb574.png)

Debug tools:
![image](https://user-images.githubusercontent.com/35329231/105864071-39232780-5ffa-11eb-8fd8-1b05ded8f218.png)

Let me know if there is any extra info needed for debugging.

**Comment by diafour** [comment]
@tambetliiv statusmap doesn't support "table" data from datasources, only "timeseries".

**Comment by tambetliiv** [comment]
Ok, thanks for the quick answer, seems like datasource provides "table". When testing with timeseries, no issues, and after changing to "table" it produces that error. 

Will try to find a way to provide data as a "timeseries"

**Comment by di97mni** [comment]
Today I tried to upgrade again. Sorry for the delay.
Grafana 7.0.0 => 7.4.3

I've set the breakpoint in `module.ts` but it's never hit

![image](https://user-images.githubusercontent.com/311406/109029641-56afe380-76c3-11eb-81a6-8ba2df2fac13.png)

Instead I see this errors in the console.log
Now I don't even come to getting any data from Azure.

![image](https://user-images.githubusercontent.com/311406/109029536-3b44d880-76c3-11eb-923a-1ffd30aa1216.png)

My javascript knowledge is very limited... 

@shurup have you had the same issue?

**Comment by di97mni** [comment]
Looking in the issue list I can see the plugin isn’t compatible with Grafana 7.4
I’ll try this again when it’s compatible

**Comment by di97mni** [comment]
Testing with 7.3.6 and using a query from the [docs](https://grafana.com/grafana/plugins/grafana-azure-data-explorer-datasource) we can see that in 7.0.0 it's a time series whereas in 7.3.6 the same query is returned as a `table`
Haven't found the reason why it, at least from 7.2.2, behaves like this

```
AzureActivity
| where $__timeFilter(TimeGenerated)
| summarize count() by Category, bin(TimeGenerated, 1h)
| order by TimeGenerated asc
```

![image](https://user-images.githubusercontent.com/311406/109058213-00eb3380-76e3-11eb-803b-581784a4b36c.png)

![image](https://user-images.githubusercontent.com/311406/109058234-05afe780-76e3-11eb-8a27-48b0b3a2dfd4.png)

**Comment by diafour** [comment]
Thank you for debugging! You've also shed the light on the mystery with Grafana 7.3.6 from https://github.com/flant/grafana-statusmap/issues/176

**Comment by di97mni** [comment]
Changing the panel type to Graph instead to make it possible to do some more debugging, otherwise the panel crash.
I then get a graph but still `table` as type in response

`Time series` is set as format
![image](https://user-images.githubusercontent.com/311406/109136301-4433b980-7758-11eb-803b-e817d524eedf.png)


![image](https://user-images.githubusercontent.com/311406/109135165-113cf600-7757-11eb-9399-787e585c0002.png)

Not sure it matters, and not due to another change like panel type, but I can also see the url is different in **7.3.6**

![image](https://user-images.githubusercontent.com/311406/109135455-62e58080-7757-11eb-9658-81ef56530f31.png)

**7.0.0**

![image](https://user-images.githubusercontent.com/311406/109135567-801a4f00-7757-11eb-94ba-a34c28f83fb4.png)

I looked in the [Grafana issues list](https://github.com/grafana/grafana/issues?q=azure+datasource) if anyone else got this issue but no luck. 
Should this issue be created in their list maybe?

**Comment by di97mni** [comment]
Changing to a query from the help section and using Graph panel we can see the same issue with `type: "table"`, even though the Graph panel supports it.

Maybe this issue should go to the [Grafana issue list](https://github.com/grafana/grafana/issues?page=3&q=azure+datasource) instead? I have looked in the list for similar issue but no success.

![image](https://user-images.githubusercontent.com/311406/109141438-05a0fd80-775e-11eb-9421-a1bfadb38bab.png)

**Comment by joshhunt** [comment]
@di97mni I'm taking a look at this from Grafana POV - it looks like you're having problems with both Azure Monitor (Logs) and ADX?

Do you have a sample query (preferably using the datatable fn?) I could run to try and dig into this? Unfortunately none of our Workspaces have AzureActivity data populated

**Comment by di97mni** [comment]
@joshhunt we are not using ADX, only log analytics and application insights.
I don't think it matters what log analytics query you are writing, we get the same issue.
I can create a sample workspace for you if you need something to query.

**Comment by joshhunt** [comment]
@di97mni Is it possible to reproduce this with the [datatable](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/datatableoperator?pivots=azuredataexplorer) operator within a Logs query with just a couple of rows? Unforunately I'm not too familiar with Statusmap and the fields it expects in the results.

Just trying to figure out exactly what's breaking down here -  it is likely that Statusmap adding support for Dataframes could fix this, but Grafana _should_ be automatically converting back to the older format anyway.

**Comment by di97mni** [comment]
I've added a docker compose and some Grafana datasource and dashboard to hopefully make it easier to find it. The dashboard contains datatable. However, due to it's not supported to use environment variables in Grafana dashboard you need to manually set WS and subscription ID in the [dashboard file](https://github.com/di97mni/Grafana-Statusmap-issue-143/blob/main/provisioning/dashboards/flant_grafana-statusmap_issues_143.json)

["REPLACE_WITH_WORKSPACE_ID"](https://github.com/di97mni/Grafana-Statusmap-issue-143/blob/main/provisioning/dashboards/flant_grafana-statusmap_issues_143.json#L74) 
["REPLACE_WITH_SUBSCRIPTION_ID"](https://github.com/di97mni/Grafana-Statusmap-issue-143/blob/main/provisioning/dashboards/flant_grafana-statusmap_issues_143.json#L90)

You also need to set the values for your Azure account in the [docker compose file](https://github.com/di97mni/Grafana-Statusmap-issue-143/blob/main/docker-compose.yml)

Then when I open panel in Grafana 7.0.0 I see this

http://localhost:3000/d/Z3lF6oQGz/flant_grafana-statusmap_issues_143?orgId=1&editPanel=2
![image](https://user-images.githubusercontent.com/311406/112017464-ed8a8700-8b2d-11eb-8cea-7279a589a81c.png)

But when open it in Grafana 7.2.2 I see this

http://localhost:3001/d/Z3lF6oQGz/flant_grafana-statusmap_issues_143?orgId=1&editPanel=2

![image](https://user-images.githubusercontent.com/311406/112017586-098e2880-8b2e-11eb-95aa-48a31c270707.png)

**Comment by di97mni** [comment]
Hi @joshhunt 
Any updates on this? Do you need more info from me?
Thanks!

**Comment by di97mni** [comment]
Probably same as #227
